services/web/server/src/simcore_service_webserver/exporter/formatters/formatter_v2.py

```diff
@@ -0,0 +1,224 @@
+from simcore_postgres_database.models.scicrunch_resources import scicrunch_resources
+from simcore_service_webserver.exporter.formatters.formatter_v1 import FormatterV1
+from simcore_service_webserver.exporter.formatters.base_formatter import BaseFormatter
+from simcore_service_webserver.exporter.formatters.sds.sds import (
+    write_sds_directory_content,
+)
+
+from simcore_service_webserver.exporter.formatters.sds.xlsx.templates.submission import (
+    SubmissionDocumentParams,
+)
+from simcore_service_webserver.exporter.formatters.sds.xlsx.templates.dataset_description import (
+    DatasetDescriptionParams,
+)
+
+from simcore_service_webserver.exporter.formatters.sds.xlsx.templates.code_description import (
+    CodeDescriptionParams,
+    CodeDescriptionModel,
+    RRIDEntry,
+    InputsEntryModel,
+    OutputsEntryModel,
+)
+
+from ..exceptions import ExporterException
+from simcore_service_webserver.projects.projects_exceptions import ProjectsException
+from simcore_service_webserver.projects.projects_api import get_project_for_user
+from simcore_service_webserver.catalog_client import get_service
+
+
+log = logging.getLogger(__name__)
+
+
+async def _get_scicrunch_resource(rrid: str, conn: SAConnection) -> Optional[RowProxy]:
+    res: ResultProxy = await conn.execute(
+        scicrunch_resources.select().where(scicrunch_resources.c.rrid == rrid)
+    )
+    return await res.first()
+
+
+async def _write_sds_content(
+    base_path: Path,
+    app: web.Application,
+    project_id: str,
+    user_id: int,
+    product_name: str,
+) -> None:
+    try:
+        project_data = await get_project_for_user(
+            app=app,
+            project_uuid=project_id,
+            user_id=user_id,
+            include_templates=True,
+            include_state=True,
+        )
+    except ProjectsException as e:
+        raise ExporterException(f"Could not find project {project_id}") from e
+
+    log.debug("Project data: %s", project_data)
+
+    # assemble params here
+    submission_params = SubmissionDocumentParams()
+    dataset_description_params = DatasetDescriptionParams(
+        name=project_data["name"], description=project_data["description"]
+    )
+
+    params_code_description = {}
+
+    rrid_entires = deque()
+
+    classifiers = project_data["classifiers"]
+    async with app[APP_DB_ENGINE_KEY].acquire() as conn:
+        for classifier in classifiers:
+            scicrunch_resource = await _get_scicrunch_resource(
+                rrid=classifier, conn=conn
+            )
+            if scicrunch_resource is None:
+                continue
+
+            rrid_entires.append(
+                RRIDEntry(
+                    rrid_term=scicrunch_resource.name,
+                    rrod_identifier=scicrunch_resource.rrid,
+                )
+            )
+    params_code_description["rrid_entires"] = list(rrid_entires)
+
+    # adding TSR data
+    quality_data = project_data["quality"]
```

Comment:
this is kind of related. In the repository you can already do validate between sqlalchemy models and pydantic models (look at all the ``...AtDB`` pydantic classes that we have! 
ONce the data is validated, you can "relax" your code and do for instance ``quality_data.enabled`` w/o checking anymore 


Same story with db error handling ...  it can be all done at the ``Repository*`` class level